plugin/src/main/java/org/opensearch/ml/sdkclient/LocalClusterIndicesClient.java

```diff
@@ -0,0 +1,127 @@
+
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * An implementation of {@link SdkClient} that stores data in a local OpenSearch cluster using the Node Client.
+ */
+@Log4j2
+public class LocalClusterIndicesClient implements SdkClient {
+
+    private final Client client;
+    private final NamedXContentRegistry xContentRegistry;
+
+    /**
+     * Instantiate this object with an OpenSearch client.
+     * @param client The client to wrap
+     * @param xContentRegistry the registry of XContent objects
+     */
+    public LocalClusterIndicesClient(Client client, NamedXContentRegistry xContentRegistry) {
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    public CompletionStage<PutDataObjectResponse> putDataObjectAsync(PutDataObjectRequest request, Executor executor) {
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<PutDataObjectResponse>) () -> {
+            try (XContentBuilder sourceBuilder = XContentFactory.jsonBuilder()) {
+                log.info("Indexing data object in {}", request.index());
+                IndexResponse indexResponse = client
+                    .index(
+                        new IndexRequest(request.index())
+                            .setRefreshPolicy(IMMEDIATE)
+                            .source(request.dataObject().toXContent(sourceBuilder, EMPTY_PARAMS))
+                    )
+                    .actionGet();
+                log.info("Creation status for id {}: {}", indexResponse.getId(), indexResponse.getResult());
+                return new PutDataObjectResponse.Builder().id(indexResponse.getId()).created(indexResponse.getResult() == CREATED).build();
+            } catch (Exception e) {
+                throw new OpenSearchException(e);
+            }
+        }), executor);
+    }
+
+    @Override
+    public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRequest request, Executor executor) {
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<GetDataObjectResponse>) () -> {
+            try {
+                log.info("Getting {} from {}", request.id(), request.index());
+                GetResponse getResponse = client.get(new GetRequest(request.index(), request.id())).actionGet();
+                if (getResponse == null || !getResponse.isExists()) {
+                    return new GetDataObjectResponse.Builder().id(request.id()).build();
+                }
+                XContentParser parser = jsonXContent
+                    .createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, getResponse.getSourceAsString());
+                log.info("Retrieved data object");
+                return new GetDataObjectResponse.Builder().id(getResponse.getId()).parser(Optional.of(parser)).build();
+            } catch (OpenSearchStatusException | IndexNotFoundException notFound) {
```

Comment:
So for `RestStatus.UNAUTHORIZED`, `RestStatus.FORBIDDEN`, `RestStatus.CONFLICT`, `RestStatus.INTERNAL_SERVER_ERROR`, `RestStatus.BAD_REQUEST`, we will throw `notFound`? This doesn't seem right to me.

Reply:
(Repeat comment on planning on aligning all exception handling with current user experience.)